Sleep manager implements all the API's in platform file

deepikabhavnani opened the pull request:
### Description

Sleep manager implements all the API's in platform/mbed_sleep_manager.h. Not sure why mbed_sleep_manager.c was in hal, moving it to platform.

### Pull request type
    [ ] Fix
    [X] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

Commit:
Sleep manager implements all the API's in platform/mbed_sleep_manager.changes

Not sure why mbed_sleep_manager.c was in hal, moving it to platform

ciarmcom commented:
@deepikabhavnani, thank you for your changes.
@ARMmbed/mbed-os-core @ARMmbed/mbed-os-maintainers please review.

mattbrown015 commented:
Do you need to update the includes after a move like this?

In other words:

    #include "platform/mbed_assert.h"
    #include "platform/mbed_power_mgmt.h"
    #include "platform/mbed_critical.h"
    #include "sleep_api.h"

should become:

    #include "mbed_assert.h"
    #include "mbed_power_mgmt.h"
    #include "mbed_critical.h"
    #include "hal/sleep_api.h"

Commit:
Update the header files

File was moved from hal -> platform folder, hence includes should be
changed accordingly.

deepikabhavnani commented:
> Do you need to update the includes after a move like this?

Thanks, updated headers 👍

deepikabhavnani commented:
@SenRamakri / @kegilbert  - Please review

deepikabhavnani commented:
Rebased on top of master to pull in travis fixes

deepikabhavnani commented:
Rebased to resolve conflicts

0xc0170 commented:
This should be ready for CI, I'll label it as such, but please can you rebase again to get latest travis fix (once it is green we start CI)

deepikabhavnani commented:
> This should be ready for CI, I'll label it as such, but please can you rebase again to get latest travis fix (once it is green we start CI)

Rebase done 👍

cmonr commented:
CI started
